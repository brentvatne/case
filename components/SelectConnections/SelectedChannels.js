import React from 'react'
import {
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from 'react-native'
import PropTypes from 'prop-types'
import { Ionicons } from '@expo/vector-icons'

import colors from '../../constants/Colors'
import layout from '../../constants/Layout'

const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
    paddingHorizontal: layout.padding,
    marginBottom: layout.padding * 3,
  },
  icon: {
    paddingRight: layout.padding / 2,
  },
  labelLine: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'center',
  },
  channelsLine: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'center',
  },
  channelWord: {
    flexDirection: 'row',
    justifyContent: 'center',
    paddingRight: layout.padding / 2,
  },
})

export default class SelectedChannels extends React.Component {
  renderChannels() {
    const { channels, onRemove } = this.props
    return channels.map((channel, index) => {
      const textColor = colors[channel.visibility]
      const showComma = index < channels.length - 1
      return (
        <TouchableOpacity key={`connection-${channel.id}`} onPress={() => onRemove(channel)}>
          <View style={styles.channelWord}>
            <Ionicons
              style={styles.icon}
              name="ios-remove-circle-outline"
              size={18}
              color={textColor}
            />
            <Text style={{ color: textColor, fontWeight: 'bold' }}>
              {channel.title}
            </Text>
            {
              showComma && <Text>,</Text>
            }
          </View>
        </TouchableOpacity>
      )
    })
  }

  render() {
    const { channels, title } = this.props
    if (channels.length) {
      return (
        <View style={styles.container}>
          <View style={styles.labelLine}>
            <Text>Connect </Text>
            <Text style={{ fontWeight: 'bold' }}>{title}</Text>
            <Text> to:</Text>
          </View>
          <View style={styles.channelsLine}>
            {this.renderChannels()}
          </View>
        </View>
      )
    }
    return (<View />)
  }
}

SelectedChannels.propTypes = {
  title: PropTypes.string,
  channels: PropTypes.any.isRequired,
  onRemove: PropTypes.func,
}

SelectedChannels.defaultProps = {
  title: 'Untitled block',
  onRemove: () => null,
}
