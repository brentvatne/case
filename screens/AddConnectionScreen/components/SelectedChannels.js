import React from 'react'
import PropTypes from 'prop-types'
import { Ionicons } from '@expo/vector-icons'

import styled from 'styled-components/native'
import { Units, Typography, Colors, Border } from '../../../constants/Style'

const Status = styled.View`
  justify-content: center;
  align-items: flex-start;
  margin-bottom: ${Units.scale[3]};
  padding-horizontal: ${Units.scale[2]};
`

const Sentence = styled.View`
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  margin-bottom: ${Units.scale[2]};
`

const StatusText = styled.Text`
  font-size: ${Typography.fontSize.base};
  color: ${Colors.gray.semiBold};
`

const BoldStatusText = styled(StatusText)`
  font-weight: ${Typography.fontWeight.semiBold};
`

const SelectedChannelsContainer = styled.View`
  flex-direction: row;
  justify-content: flex-start;
  flex-wrap: wrap;
`

const SelectedChannel = styled.TouchableOpacity`
  margin-right: ${Units.scale[2]};
  margin-vertical: ${Units.scale[1]};
  border-width: ${Border.borderWidth};
  border-radius: ${Border.borderRadius};
  padding-horizontal: ${Units.scale[2]};
  padding-vertical: ${Units.scale[1]};
  flex-direction: row;
  align-items: center;
`

const Icon = styled(Ionicons)`
  padding-right: ${Units.scale[1]};
  position: relative;
  bottom: -1;
`

const ChannelWord = styled.Text`
  font-weight: ${Typography.fontWeight.semiBold};
`

export default class SelectedChannels extends React.Component {
  renderChannels() {
    const { channels, onRemove } = this.props
    return channels.map((channel) => {
      const channelColor = Colors.channel[channel.visibility]
      const backgroundColor = Colors.channel.background[channel.visibility]

      return (
        <SelectedChannel
          key={`connection-${channel.id}`}
          onPress={() => onRemove(channel)}
          style={{ backgroundColor, borderColor: channelColor }}
        >
          <Icon
            name="ios-close-outline"
            size={20}
            color={channelColor}
          />
          <ChannelWord style={{ color: channelColor }}>
            {channel.title}
          </ChannelWord>
        </SelectedChannel>
      )
    })
  }

  render() {
    const { channels, title, isSearching } = this.props
    if (channels.length > 0 || isSearching) {
      return (
        <Status>
          <Sentence>
            <StatusText>
              <StatusText>Connect </StatusText>
              <BoldStatusText>&#8220;{title}&#8220;</BoldStatusText>
              <StatusText> to</StatusText>
            </StatusText>
          </Sentence>
          {channels.length > 0 &&
            <SelectedChannelsContainer>
              {this.renderChannels()}
            </SelectedChannelsContainer>
          }
        </Status>
      )
    }
    return (
      <Status>
        <Sentence>
          <StatusText>
            Recent channels
          </StatusText>
        </Sentence>
      </Status>
    )
  }
}

SelectedChannels.propTypes = {
  title: PropTypes.string,
  channels: PropTypes.any.isRequired,
  onRemove: PropTypes.func,
  isSearching: PropTypes.bool,
}

SelectedChannels.defaultProps = {
  title: 'Untitled block',
  onRemove: () => null,
  isSearching: false,
}
