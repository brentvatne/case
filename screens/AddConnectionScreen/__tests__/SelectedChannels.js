import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import SelectedChannels from '../components/SelectedChannels'

const channelUno = {
  title: 'Round things',
  visibility: 'private',
}
const channelDos = {
  title: 'Square things',
  visibility: 'private',
}

xtest('renders correctly', () => {
  const tree = shallow(
    <SelectedChannels channels={[channelUno, channelDos]} />,
  )
  expect(toJSON(tree)).toMatchSnapshot()
})
