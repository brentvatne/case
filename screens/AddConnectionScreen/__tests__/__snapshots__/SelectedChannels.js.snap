// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Styled(View)>
  <Styled(View)>
    <Styled(Text)>
      <Styled(Text)>
        Connect 
      </Styled(Text)>
      <Styled(Styled(Text))>
        “
        Untitled block
        “
      </Styled(Styled(Text))>
      <Styled(Text)>
         to
      </Styled(Text)>
    </Styled(Text)>
  </Styled(View)>
  <Styled(View)>
    <Styled(TouchableOpacity)
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "rgb(250, 242, 242)",
          "borderColor": "rgb(165, 0, 0)",
        }
      }
    >
      <Styled(Icon)
        color="rgb(165, 0, 0)"
        name="ios-close-outline"
        size={20}
      />
      <Styled(Text)
        style={
          Object {
            "color": "rgb(165, 0, 0)",
          }
        }
      >
        Round things
      </Styled(Text)>
    </Styled(TouchableOpacity)>
    <Styled(TouchableOpacity)
      onPress={[Function]}
      style={
        Object {
          "backgroundColor": "rgb(250, 242, 242)",
          "borderColor": "rgb(165, 0, 0)",
        }
      }
    >
      <Styled(Icon)
        color="rgb(165, 0, 0)"
        name="ios-close-outline"
        size={20}
      />
      <Styled(Text)
        style={
          Object {
            "color": "rgb(165, 0, 0)",
          }
        }
      >
        Square things
      </Styled(Text)>
    </Styled(TouchableOpacity)>
  </Styled(View)>
</Styled(View)>
`;
